cmd/mr_checkout.go

```diff
@@ -71,18 +70,37 @@ var checkoutCmd = &cobra.Command{
 		// the fetchToRef to the mr author/sourceBranch
 		if mrCheckoutCfg.track {
 			// Check if remote already exists
-			if _, err := gitconfig.Local("remote." + mr.Author.Username + ".url"); err != nil {
-				// Find and create remote
-				mrProject, err := lab.GetProject(mr.SourceProjectID)
+			project, err := lab.GetProject(mr.SourceProjectID)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			remotes, err := git.Remotes()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			remoteName := ""
+			for _, remote := range remotes {
+				path, err := git.PathWithNamespace(remote)
 				if err != nil {
-					log.Fatal(err)
+					continue
 				}
-				urlToRepo := labURLToRepo(mrProject)
-				if err := git.RemoteAdd(mr.Author.Username, urlToRepo, "."); err != nil {
+				if path == project.PathWithNamespace {
+					remoteName = remote
+				}
```

Comment:
Do you think it's common for folks to have multiple remotes with the same `PathWithNamespace`? If so, would it make sense to always use the first matching remote? ie to `break` from the loop when a matching repo is found? As written, this will always use the last one that matched, I think.

I suppose it will be hard to have a deterministic result here if the enduser is adding remotes that all point to the same place. The remote that's chosen will depend on the order of remotes returned by `git remote`, which I think (based on exactly 1 test 😄) is sorted alphabetically.

Reply:
Well pointed. It should have a `break` in there for sure. If the user wants to have numerous remotes pointing to the same project path that's not `lab` responsibility to clean things up or guess things further 😅 IMHO.

Reply:
Fixed.